Source/Initialization/DivCleaner/ProjectionDivCleaner.cpp

```diff
@@ -288,34 +359,53 @@ ProjectionDivCleaner::correctBfield ()
                 Bz_arr(i,j,0) += CylindricalYeeAlgorithm::DownwardDz(sol_arr, coefs_z, n_coefs_z, i, j, 0, 0);
             });
 #else
-            amrex::ParallelFor(tbx, tby, tbz,
-            [=] AMREX_GPU_DEVICE (int i, int j, int k)
-            {
-                Bx_arr(i,j,k) += CartesianYeeAlgorithm::DownwardDx(sol_arr, coefs_x, n_coefs_x, i, j, k);
-            },
-            [=] AMREX_GPU_DEVICE (int i, int j, int k)
-            {
-                By_arr(i,j,k) += CartesianYeeAlgorithm::DownwardDy(sol_arr, coefs_y, n_coefs_y, i, j, k);
-            },
-            [=] AMREX_GPU_DEVICE (int i, int j, int k)
-            {
-                Bz_arr(i,j,k) += CartesianYeeAlgorithm::DownwardDz(sol_arr, coefs_z, n_coefs_z, i, j, k);
-            });
+            if (m_grid_type == GridType::Staggered) {
```

Comment:
Perhaps this function can also be simpified by templating it on the finite difference algorithm?